### DISCUSSION TITLE
Esphome media player - how do I add that to music assistant

### DISCUSSION THREAD

**Discussion by alfwro13** [music-assistant/support#1207] [Q&A] [answered]
I have an ESPhome media player. Outside Music Assistant Beta i.e. via HA media player I can play music to that device without any issues. Also in the old version of  Music Assistant it used to work as well. However in the Add-on version of the Music Assistant for the life of me I cannot figure out how to add my ESPhome based media player. All I can see is my google speaker. Can someone please point me in the right direction.
Thank you

**Comment by OzGav** [comment]
Only certain player types are supported by v2. When the HA integration is written all the HA media player entities will be exposed. However, possibly not all MA features will be available if there isn’t a native MA player provider. Vote or add to the player provider poll to indicate which players you would like native support for. (See the pinned posts for the poll)

**Reply by OzGav** [reply, marked as answer]
Answered here https://github.com/orgs/music-assistant/discussions/1338
